src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -387,17 +387,18 @@ def load(self,
 
         schedule_cfg = self.handler.module.get_module_option(
             self.handler.MODULE_OPTION_NAME, '')
+        schedule_local_cfg = self.handler.module.get_localized_module_option(
+             self.handler.MODULE_OPTION_NAME, '')
 
         # Previous versions incorrectly stored the global config in
         # the localized module option. Check the config is here and fix it.
-        if not schedule_cfg:
-            schedule_cfg = self.handler.module.get_localized_module_option(
-                self.handler.MODULE_OPTION_NAME, '')
-            if schedule_cfg:
+        if schedule_local_cfg:
+            self.handler.module.set_localized_module_option(
+                self.handler.MODULE_OPTION_NAME, None)
+            if not schedule_cfg:
+                schedule_cfg = schedule_local_cfg
                 self.handler.module.set_module_option(
                     self.handler.MODULE_OPTION_NAME, schedule_cfg)
```

Comment:
nit: It's really highly unlikely, but still there is a chance the mgr crashes between removing the localized option and setting the global option. So I would prefer we removed the localized option only after setting the global.

Reply:
good point. done.